### PR TITLE
Update FE2 documentation

### PR DESCRIPTION
**Pull request by matteosz**
This PR updates the doc with the work of this semester

### CODE REVIEW

**Review by simone-kalbermatter — APPROVED** [review]
LGTM!

File: fe2-android/docs/README.md [hunk 1]
```diff
@@ -324,17 +278,35 @@ This solution is equivalent to `provideDatabaseImpl()` and it is widely used in
...
+
+One of the most important things to keep in mind is to perform the database operations in an asynchronous fashion and preferably using an I/O thread. This is easily possible thanks to the support for `RxJava2` by the `Room` library, such that the methods in the DAOs could return a `Completable` or a `Single`. However, having asynchronous queries is not always possible, in fact very few queries are performed on the main thread but this approach should be as limited as possible.
+
+There's no a default behaviour of storing and retrieving the persistent state of a repository, so here's more details:
```
Comment (Warhedgehog) [inline fe2-android/docs/README.md:289]: Typo :)

**Review by Warhedgehog — CHANGES_REQUESTED** [review]
Great work. Just the typo

File: fe2-android/docs/README.md (hunk 1 above)
Comment (Warhedgehog) [reply fe2-android/docs/README.md:289]: Sorry it's at the start of line 289

**Review by Warhedgehog — APPROVED** [review]
LGTM

**Pull request merged**